alerter.c

```diff
@@ -1,31 +1,42 @@
 #include <stdio.h>
 #include <assert.h>
+#include "alerter_test.h"

```

Comment:
File containing production code also contains test code and depends on test file too... this kind of dependency should be avoided. Test code should depend on or use production code and not vice versa. 
Code can be rearranged ...